Source/WebCore/editing/FormatBlockCommand.cpp

```diff
@@ -99,6 +100,10 @@ void FormatBlockCommand::formatRange(const Position& start, const Position& end,
 
     moveParagraphWithClones(start, end, blockNode.get(), outerBlock.get());
 
+    // Copy the inline style of the original block element to the newly created block-style element.
+    if (outerBlock.get() != nodeAfterInsertionPosition.get() && downcast<HTMLElement>(nodeAfterInsertionPosition.get())->hasAttribute(styleAttr))
```

Comment:
Why are we checking outerBlock here but then setting style attribute on blockNode?

Reply:
OuterBlock is checked to ensure that any inherited style from ancestor is also copied and then applied to blockNode (two approaches possible - this current approach but alternative is avoided - which is to add another style span around replaced block element (similar to commit message)).

Reply:
Why are we not checking blockNode.get() != nodeAfterInsertionPosition.get() instead?

Reply:
From my understanding, OuterBlock will cover all nodes and will be able to traverse through them while blockNode will be limited to nodes which are within Block Element. I am happy to modify and run it through EWS and happy as long as the test case passes and there is no other issues. Happy to be correct in learning something new or if I reached to wrong conclusion?

Reply:
I don't think the code change as currently proposed makes much sense. For one, there is mismatch between outerBlock vs blockNode. For another, we shouldn't be just copying style attribute. We need to get the computed style and apply that style to the block if we're failing to preserve the style in this command. e.g. using EditingStyle::prepareToApplyAt and ApplyStyleCommand.

Reply:
Make sense - I think point in time, it would be difficult to tackle this bug for me. Let me close this PR and read "prepareToApplyAt" and AppleStyleCommand and see how we can use it.